src/clj/rems/application/commands.clj

```diff
@@ -263,6 +259,23 @@
 (defn- ok [& events]
   (ok-with-data nil events))
 
+(defn- invalid-attachments-error [injections cmd]
+  (let [invalid-ids (for [id (:attachments cmd)
+                          :let [attachment ((:get-attachment-metadata injections) id)]
```

Comment:
I think we could start using the `getx` pattern here (or introduce `callx`).

Reply:
good point, I'll add a getx